### ISSUE TITLE
AttributeError: 'list' object has no attribute 'keys'

### ISSUE THREAD

**Issue by tych777** [ccxt-dev/ccxt#246]
```python
import ccxt.async as ccxt
import asyncio
polo = ccxt.poloniex(({
    'apiKey': '4ZY5Zbbbbbbbbbb11WUV',
    'secret': 'e0c68650e32bbbbbbbb024c5a950dbf659eaaaaaaaad1423bbbbbbbbc77620c0bc3cb3c7a3d7042a9baf45',
    'verbose': True
}))
asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades())
```

gives me the following error

```
https://poloniex.com/public?command=returnTicker GET https://poloniex.com/public?command=returnTicker 
Request: {'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'ccxt/1.8.70 (+https://github.com/kroitor/ccxt) Python/3.5.3'} None
GET https://poloniex.com/public?command=returnTicker 
Response: {'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'ccxt/1.8.70 (+https://github.com/kroitor/ccxt) Python/3.5.3'} {"BTC_BCN":{"id":7,"last":"0.00000031","lowestAsk":"0.00000032","highestBid":"0.00000031","percentChange":"-0.11428571","baseVolume":"72.88360975","quoteVolume":"219789281.59626094","isFrozen":"0","high24hr":"0.00000035","low24hr":"0.00000031"}................................................................................................................................................................."ETH_GAS{"id":199,"last":"0.07480000","lowestAsk":"0.07721282","highestBid":"0.07480000","percentChange":"-0.08442820","baseVolume":"360.61812422","quoteVolume":"4577.44707171","isFrozen":"0","high24hr":"0.08534279","low24hr":"0.07442987"}}
https://poloniex.com/tradingApi POST https://poloniex.com/tradingApi 
Request: {'Key': '4xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxV', 'Sign': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbccccccccccc', 'User-Agent': 'ccxt/1.8.70 (+https://github.com/kroitor/ccxt) Python/3.5.3', 'Accept-Encoding': 'gzip, deflate', 'Content-Type': 'application/x-www-form-urlencoded'} end=1506669048&nonce=1506669048&currencyPair=all&command=returnTradeHistory
POST https://poloniex.com/tradingApi 
Response: {'Key': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', 'Sign': 'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb', 'User-Agent': 'ccxt/1.8.70 (+https://github.com/kroitor/ccxt) Python/3.5.3', 'Accept-Encoding': 'gzip, deflate', 'Content-Type': 'application/x-www-form-urlencoded'} []
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-42-13a73354e37f> in <module>()
----> 1 asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades())

C:\Users\ggg\Anaconda3\lib\asyncio\base_events.py in run_until_complete(self, future)
    464             raise RuntimeError('Event loop stopped before Future completed.')
    465 
--> 466         return future.result()
    467 
    468     def stop(self):

C:\Users\ggg\Anaconda3\lib\asyncio\futures.py in result(self)
    291             self._tb_logger = None
    292         if self._exception is not None:
--> 293             raise self._exception
    294         return self._result
    295 

C:\Users\ggg\Anaconda3\lib\asyncio\tasks.py in _step(***failed resolving arguments***)
    237                 # We use the `send` method directly, because coroutines
    238                 # don't have `__iter__` and `__next__` methods.
--> 239                 result = coro.send(None)
    240             else:
    241                 result = coro.throw(exc)

C:\Users\ggg\Anaconda3\lib\site-packages\ccxt\async\exchanges.py in fetch_my_trades(self, symbol, params)
  15360         else:
  15361             result = {'info': response}
> 15362             ids = list(response.keys())
  15363             for i in range(0, len(ids)):
  15364                 id = ids[i]

AttributeError: 'list' object has no attribute 'keys'
```

Why am I getting this error?
Thanks

**Comment by kroitor** [comment]
Hi! I'll take a look and report back to you here asap.

**Comment by kroitor** [comment]
Ok, I uploaded a minor fix for it, please update to latest version of ccxt and try this:

```Python
import ccxt.async as ccxt
import asyncio
polo = ccxt.poloniex(({
    "apiKey": "YOUR_API_KEY",
    "secret": "YOUR_SECRET",
    "verbose": True
}))

# fetch all trades for last 24 hours
asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades())

# polo outputs one day by default
# use additional params to get older trades
params = { 'start': polo.seconds() - 86400 * 7 } # a week (7 days) ago

# fetch all trades for the last week
# asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades(None, params))

# fetch BTC/USDT trades for the last week
# asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades('BTC/USDT', params))
```

Note, that Poloniex will return only last 24 hours of trades if you don't specify any additional params (start, end) to it, it's a limitation built into their API.
Does it work for you?

**Comment by kroitor** [comment]
Ok, I uploaded a minor fix for it, please update to version 1.8.72+ and try this:

```
import ccxt.async as ccxt
import asyncio
polo = ccxt.poloniex(({
    "apiKey": "YOUR_API_KEY",
    "secret": "YOUR_SECRET",
    "verbose": True
}))

# fetch all trades for last 24 hours
asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades())

# polo outputs one day by default
# use additional params to get older trades
params = { 'start': polo.seconds() - 86400 * 7 } # a week (7 days) ago

# fetch all trades for the last week
# asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades(None, params))

# fetch BTC/USDT trades for the last week
# asyncio.get_event_loop().run_until_complete(polo.fetch_my_trades('BTC/USDT', params))
```

Note, that Poloniex will return only last 24 hours of trades if you don't specify any additional params (start, end) to it, it's a limitation built into their API.
